controllers/servicebinding_controller.go

```diff
@@ -651,6 +686,14 @@ func (r *ServiceBindingReconciler) singleKeyMap(credentialsMap map[string][]byte
 	}, nil
 }
 
+func (r *ServiceBindingReconciler) rotateCredentials(ctx context.Context, smClient sm.Client, binding *v1alpha1.ServiceBinding) {
```

Comment:
I am wondering how this signature can be implemented locally. The full lifecycle for rebinding should involve that workloads pick up new credentials before the old credentials get deleted via unbind. Is there a design document giving more clarity on how this will work and what constraints it will require the workloads to adhere to?

Reply:
Hi,
This is still WIP, please don't review the PR yet this BLI planned for 2A.
The idea is to rename the original binding directly in SM and "trick" the K8S object to "think" that binding never created, and it will create the new binding and override his secret value.

Workloads won't be affected as the old binding still valid until next rotation.